Assets/Scripts/Player/Input Parser/InputParser.cs

```diff
@@ -1,22 +1,41 @@
+using System;
 using Player.Movement;
 using UnityEngine;
 using UnityEngine.InputSystem;
+using FrameWork;
 
 namespace Player.Input_Parser
 {
     public sealed class InputParser : MonoBehaviour
     {
+        [SerializeField, Range(0, 20)] private float interactableRayDistance;
+        
         private PlayerInput _playerInput;
         private InputActionAsset _inputActionAsset;
         private PlayerMovement _playerMovement;
 
+        private Camera _mainCamera;
+        private InteractableObject _lastObject;
```

Comment:
```suggestion
        private InteractableObject _lastInteractable;
```
`_lastObject` kan elke object zijn.